app/controllers/programs_controller.rb

```diff
@@ -6,4 +6,30 @@ def index
   def show
     @program = Program.find(params[:id])
   end
+
+  def create
+    @program = Program.new(program_params)
+
+    if @program.save
+      flash[:notice] = "Program saved"
+      redirect_to program_path(@program)
+    else
+      flash[:notice] = "Fill out form correctly"
```

Comment:
Rather than using a generic message, collect the validation errors on the object and display them.